### ISSUE TITLE
Add a pinball loss metric and deprecate rho_risk (?)

### ISSUE THREAD

**Issue by deltamacht** [unit8co/darts#1367]
I've been playing around with some probabilistic predictions with various models. I've done an LGBM quantile regression as well as a TFT model with probabilistic outputs. In both cases I computed the quantiles [0.05, 0.5, 0.95]. When I view the output I find that the 0.05 and 0.5 quantiles are pretty similar between the two models, whereas the LGBM's 0.95 quantile is much, much higher than that of the TFT (and is generally a lot higher than the actual data). However, when I computed the rho_risk for quantiles 0.05, 0.5, and 0.95, I find that the two models have similar rho risk for 0.5 and 0.95, but much different rho risk for 0.05 (so, the opposite of what I would have expected looking at the plots).

This led me down the path of trying to figure out how this could be, and looking at the rho_risk code itself. While it seems like the implementation of rho_risk is according to the description in the paper that went with the PR for this feature, I don't really understand why it's supposed to work. 

It's a simple bit of code, so I'll include it here for brevity:
  
```
z_true = z_true.sum(axis=0)
z_hat = z_hat.sum(axis=0)  # aggregate all individual sample realizations

z_hat_rho = np.quantile(z_hat, q=rho)  # get the quantile from aggregated samples

pred_above = np.where(z_hat_rho >= z_true, 1, 0)
pred_below = np.where(z_hat_rho < z_true, 1, 0)

rho_loss = 2 * (z_true - z_hat_rho) * (rho * pred_below - (1 - rho) * pred_above)
return rho_loss / z_true
```

Let's imagine an artificial time zeries `z_true`, which alternates between -1 and 1 on subsequent time steps. As such, it's `z_true.sum(axis=0)` is 0. Now let's imagine a `z_hat` which alternates from 1 and -1 on subsequent time steps (either 1 sample or you could sample from a gaussian of small width to make it stochastic; let's say you sample 10 samples from a gaussian of width 0.01 or something). I would expect `z_hat_rho` for the 0.5 quantile to be 0, and thus `z_true - z_hat_rho` would also be 0, and thus the rho_loss would be 0 which corresponds to a perfect score. However, for this hypothetical series where the predictions are out of phase with the targets by one time step (being -1 when the target is 1, and being 1 when the target is -1), the target is never even within the sampled distribution, so how could it get such a perfect score?

Am I thinking of this incorrectly? Also want to note that there's a SO question from earlier this year where another user is questioning whether the rho_risk values they are getting make sense: https://datascience.stackexchange.com/questions/108892/how-can-i-interpret-my-rho-risk-values-when-performing-probabilistic-time-series.

**Comment by hrzn** [comment]
Hi @deltamacht, you're right, looking at this equation again, I can't seem to really make sense of the `(z_true - z_hat_rho)` term. There might be more to it that I'm also not seeing... however I'm thinking most of the time we should probably consider using the quantile loss (aka pinball loss) instead of rho risk. I'll move this ticket to our backlog so we add this function and consider diving deeper into rho risk to see whether it makes sense to keep it or not.

**Comment by JanFidor** [comment]
@hrzn Hi, it's me again could pick this up? Just the straight forward addition of the pinball loss metric, without the deep diving ;)

**Comment by hrzn** [comment]
@JanFidor that'd be great! Thanks :)
